code/game/machinery/vending/vendor_types/antag/antag_predator.dm

```diff
@@ -98,6 +123,11 @@ GLOBAL_LIST_INIT(cm_vending_elder_yautja, list(
 			vend_fail()
 		return FALSE
 
+	if(user.faction == FACTION_YAUTJA_YOUNG)
```

Comment:
Put this logic under `allowed()` (override the proc and make sure to call the parent) and return false on anything other than `FACTION_YAUTJA` rather than checking for youngblood